### ISSUE TITLE
Package not available in maven central repo

### ISSUE THREAD

**Issue by kapil-gahlawat** [brightcove/Adobe-AEM-Brightcove-Connector#53]
Hi,

We are using Brightcove and integrated it with AEM. As with most of the projects these days, we are also using maven for build and deployment. With this Brightcove AEM connector, right now we need to install it manually to our AEM. We want to automate this deployment but problem is that this AEM-Brightcove connector is not available in a public repo (Maven central repo). So, we can't automate the deployment of this connector package with our code. It is problematic especially for AMS and AEM as a Cloud service as it restricts doing anything manually in AEM servers.

The only solution is to set up our own Nexus and host this connector package there. But there are a couple of problems with this approach:
1.)  If the company does not have nexus, they need to set up and manage this server just to host 1 dependency.
2.) If a company already have their own Nexus, it is usually internal and can be accessed only through a VPN. As you know AEM is now hosted mostly with AMS (Adobe Managed Services) or AEMaaCS (AEM as a Cloud Service). In both cases, they can't access our VPN-enabled Nexus. We can't make our Nexus public as well for security reasons.

So, wouldn't it be better if you guys can publish this connector to Maven central repo? It is a simple and straightforward task for you guys and it will save lots of effort and time for most people.

P.S: We are using AEM Brightcove connector version 5.6.1 right now.

Thanks,
Kapil.

**Comment by hngrylobster** [comment]
Hi Kapil,

We do offer a Nexus repository for those who would prefer not to host it themselves. You can use the following:

```xml
<repository>
    <id>central</id>
    <name>Brightcove Repo</name>
    <url>https://nexus.3sharecorp.com</url>
    <layout>default</layout>
    <snapshots>
        <enabled>false</enabled>
    </snapshots>
</repository>
```

```xml
<dependency>
  <groupId>com.brightcove</groupId>
  <artifactId>com.brightcove.connector</artifactId>
  <version>6.0</version>
</dependency>
```

That said, Brightcove does not offer any sort of guarantee that this will be accessible indefinitely.

Thanks,
Paul

**Comment by kapil-gahlawat** [comment]
@hngrylobster Thanks for the information, Paul. is this hosted version for cloud or on-prem?

**Comment by hngrylobster** [comment]
Cloud-only at the moment.

**Comment by kapil-gahlawat** [comment]
@hngrylobster Could you please also host on-prem version as well? We are on AMS, so would need an on-prem version.

**Comment by hngrylobster** [comment]
It is possible to use private artifact servers with AMS/Cloud installations. That said, we might be able to upload this to our repository but it won't be until after the holidays, unfortunately.

**Comment by hngrylobster** [comment]
We are working with Brightcove on a new release so this will happen soon (later in January).

**Comment by kapil-gahlawat** [comment]
@hngrylobster : Even the 6.0 version that is hosted in 3share repo is not complete. It is just 300 kb and does not contain the bundle. It just contains /apps/components, /apps/templates, /apps/extensions and /apps/clientlibs. It does not contain configs and the core bundle.

**Comment by hngrylobster** [comment]
Can you confirm which version you are pointing to? Our Nexus artifact appears like this:

<img width="404" alt="image" src="https://user-images.githubusercontent.com/7432966/230112378-77337b09-3b32-4507-9483-eebfa2b7c6ba.png">

**Comment by kapil-gahlawat** [comment]
@hngrylobster Looks like it was added to your artefact today. We were earlier pointing to this: https://nexus.3sharecorp.com/#browse/search=keyword%3Dcom.brightcove:f21ff3b2a4c24aaa7c025d535e16c2ce

Also, the cloud manager build is failing at the "Build Images" step for both 5.7 and 6.0. We used them after downloading from here: https://github.com/brightcove/Adobe-AEM-Brightcove-Connector/releases/tag/6.0.0-cloud. Already reported it in another issue: https://github.com/brightcove/Adobe-AEM-Brightcove-Connector/issues/56

**Comment by hngrylobster** [comment]
Hi - is this still an issue with the latest package? We have seen this before and Adobe provided a recommendation to another client for how the embedded packaging needs to work due to a limitation in Cloud Manager. If this is still pending, have you reached out to Adobe at all yet regarding this issue?

**Comment by kapil-gahlawat** [comment]
Hi,
This issue has been resolved now. We had to make some changes in your code and rebuild a new package. It was Cloud manager specific issue.

The issue was that we were installing the Brightcove package only in the author instance using install.author folder embedding. But Brightcove itself contains other modules like core, apps etc which are using the install folder. Cloud manager was ignoring this hierarchy. It was trying to install Brightcove in author but its submodules in publish as well. So, there was some inconsistency and the build was failing. We had to make changes to module embedding and used install.author for submodels as well. We recompiled your code and deployed and it worked.
